src/Lumina/Data/Parsing/MdlStructs.cs

```diff
@@ -0,0 +1,397 @@
+                ret.Type = br.ReadByte();
+                ret.Usage = br.ReadByte();
+                ret.UsageIndex = br.ReadByte();
+                ret.Padding = br.ReadBytes( 3 );
+                return ret;
+            }
+        }
+
+        public struct ModelHeader
+        {
+            // MeshHeader
+            public float Radius;
+            public ushort MeshCount;
+            public ushort AttributeCount;
+            public ushort SubmeshCount;
+            public ushort MaterialCount;
+            public ushort BoneCount;
+            public ushort BoneTableCount;
+            public ushort ShapeCount;
+            public ushort ShapeMeshCount;
+            public ushort ShapeValueCount;
+            public byte LodCount;
+
+            private byte bitfield1;
+
+            public bool DustOcclusionEnabled;
+            public bool SnowOcclusionEnabled;
+            public bool RainOcclusionEnabled;
+            public bool Unknown1;
+            public bool BGLightingReflectionEnabled;
+            public bool WavingAnimationDisabled;
+            public bool LightShadowDisabled;
+            public bool ShadowDisabled;
+            public ushort ElementIdCount;
+            public byte TerrainShadowMeshCount;
+
+            private byte bitfield2;
+
+            public bool Unknown2;
+            public bool BgUvScrollEnabled;
+            public bool EnableForceNonResident;
+            public bool ExtraLodEnabled;
+            public bool ShadowMaskEnabled;
+            public bool ForceLodRangeEnabled;
+            public bool EdgeGeometryEnabled;
+            public bool Unknown3;
+
+            public float ModelClipOutDistance;
+            public float ShadowClipOutDistance;
+            public ushort Unknown4;
+            public ushort TerrainShadowSubmeshCount;
+
+            private byte Unknown5;
+
+            public byte BGChangeMaterialIndex;
+            public byte BGCrestChangeMaterialIndex;
+            public byte Unknown6;
+            public ushort Unknown7;
+            public ushort Unknown8;
+            public ushort Unknown9;
+            private byte[] Padding00;
+
+            public static ModelHeader Read( BinaryReader br )
+            {
+                ModelHeader ret = new ModelHeader();
+                ret.Radius = br.ReadSingle();
+                ret.MeshCount = br.ReadUInt16();
+                ret.AttributeCount = br.ReadUInt16();
+                ret.SubmeshCount = br.ReadUInt16();
+                ret.MaterialCount = br.ReadUInt16();
+                ret.BoneCount = br.ReadUInt16();
+                ret.BoneTableCount = br.ReadUInt16();
+                ret.ShapeCount = br.ReadUInt16();
+                ret.ShapeMeshCount = br.ReadUInt16();
+                ret.ShapeValueCount = br.ReadUInt16();
+                ret.LodCount = br.ReadByte();
+
+                ret.bitfield1 = br.ReadByte();
+                ret.DustOcclusionEnabled = ( ret.bitfield1 & 0x80 ) == 0x80;
```

Comment:
particular reason why this and the one below doesn't just use an enum instead?

Reply:
resolved w/ [Flag] enums and auto-property accessors